profiles/common/modules/custom/ecas/includes/ecas.inc

```diff
@@ -351,23 +349,66 @@ function ecas_sync_profile_core(&$user, array $user_info) {
  *   info_ecas_update() hook.
  */
 function ecas_sync_user_info(&$user, array $user_info, array $args) {
+  $edit = array();
   // Update the user mail in the users table.
   if (variable_get('ecas_update_mail_address', TRUE)) {
-    $picture = $user->picture;
-    // Fix : save file object replaced by fid by user_save function.
-    user_save($user, array('mail' => $user_info['cas:email']));
-    $user->mail = $user_info['cas:email'];
-    $user->picture = $picture;
+    $edit['mail'] = $user_info['cas:email'];
   }
 
   // Update extra core user fields.
   if (variable_get('ecas_profile_core_enabled', FALSE)) {
-    ecas_sync_profile_core($user, $user_info);
+    ecas_sync_profile_core($user, $user_info, $edit);
+  }
+
+  // Call "hook_ecas_sync_user_info" implementation with $edit passed by
+  // reference as parameter.
+  foreach (module_implements('ecas_sync_user_info') as $module) {
+    $function = $module . '_ecas_sync_user_info';
+    $function($user, $user_info, $edit, $args);
   }
 
+  // Let the opportunity to modules to alter user roles.
+  drupal_alter('ecas_user_roles', $edit['roles'], $user, $user_info, $args);
+
+  // Fix : save file object replaced by fid by user_save function.
+  $picture = $user->picture;
+  user_save($user, $edit);
+  $user->mail = $user_info['cas:email'];
+  $user->picture = $picture;
+
+  // Old hook implementation. we keep it for backward compatibility.
   module_invoke_all('info_ecas_update', $user, $user_info, $args);
 }
 
+/**
+ * Implements hook_ecas_sync_user_info().
+ *
+ * Assign the "CEM" role assigned to the user if they belong to the COMM_CEM.
+ */
+function ecas_ecas_sync_user_info($user, array $user_info, array &$edit, array $args) {
+  $cem_group = 'COMM_CEM';
+  $cem_role_name = 'cem';
```

Comment:
'COMM_CEM' and 'cem' strings should be constants.
We already have a role name defined in a constant: NEXTEUROPA_EDITORIAL_TEAM_MEMBER_ROLE.